src/validation.cpp

```diff
@@ -1851,6 +1851,18 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // SEGWIT2X signalling.
+    if ( VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE &&
```

Comment:
Activation before August 1 is already infeasible and not part of the agreement.

Removing the LOCKED_IN sets a terrible precedent as every update is voluntary. You would completely remove the upgrade window for miners who disagree but don't want to lose money by  not signalling until LOCKED_IN.